packages/palette/src/elements/Select/Select.tsx

```diff
@@ -51,6 +57,50 @@ export const Select: ForwardRefExoticComponent<
     ref
   ) => {
     const [boxProps, selectProps] = splitBoxProps(rest)
+    // due to :has not available in Firefox yet, we need to add the styles to the label using JS
+    const [selectedOption, setSelectedOption] = useState(selected)
+    const [isFocused, setIsFocused] = useState(false)
+    const [isHovered, setIsHovered] = useState(false)
+
+    // if there is no selected option (or is it empty)
+    // and the select is not focused
+    // and the title prop is present
+    const selectStyle = !isFocused &&
+      !selectedOption &&
+      !!title && {
+        color: "transparent",
+      }
+
+    // if there is a selected option
+    // and the select is not focused
+    // and the title prop is present
+    const labelSpanStyle = !isFocused &&
+      !!selectedOption &&
+      !!title && {
+        height: 2,
+        top: "50%",
+      }
+
+    // if the selected option is not empty
+    const labelStyleSelected = selectedOption && {
+      transform: "translateY(-150%)",
+      fontSize: THEME.textVariants.xs.fontSize,
+    }
+
+    // if the select is focused and there is no selected option (or it is empty)
+    const labelStyleFocused = isFocused &&
+      !selectedOption && {
+        textDecoration: "underline",
+      }
+
+    // if the select is hovered and there is no selected option (or it is empty)
+    // and the select is not disabled
+    const labelStyleHovered = isHovered &&
+      !selectedOption &&
+      !disabled && {
+        textDecoration: "underline",
+        color: THEME.colors.blue100,
+      }
```

Comment:
A few things:
* Why not leave these declarations in the tokens file and just use them here?
* You don't want to be accessing the `THEME` object — just spread the props on to the label, don't use `style`

Reply:
> You don't want to be accessing the THEME object — just spread the props on to the label, don't use style

I don't think I can apply the styles by spreading the props on the components, I tested it and it wasn't working, especially with `<span>`, I'm definitely missing something but I'm not sure what, would love your input on this!

Reply:
Sure you can: `<Box as="span">`